src/storage/src/hummock/event_handler/uploader.rs

```diff
@@ -858,46 +971,39 @@ impl HummockUploader {
                 .stats
                 .uploader_syncing_epoch_count
                 .set(self.syncing_data.len() as _);
-            let epoch = syncing_data.sync_epoch();
-
-            let newly_uploaded_sstable_infos = match &result {
-                Ok(sstable_infos) => sstable_infos.clone(),
-                Err(_) => vec![],
-            };
+            let epoch = syncing_data.sync_epoch;
 
-            let result = result.map(|mut sstable_infos| {
-                // The newly uploaded `sstable_infos` contains newer data. Therefore,
-                // `sstable_infos` at the front
-                sstable_infos.extend(syncing_data.uploaded);
+            let result = result.map(|newly_uploaded_sstable_infos| {
+                // take `rev` so that old data is acked first
+                for sstable_info in newly_uploaded_sstable_infos.iter().rev() {
+                    self.unsync_data.ack_flushed(sstable_info);
+                }
                 SyncedData {
-                    staging_ssts: sstable_infos,
+                    newly_upload_ssts: newly_uploaded_sstable_infos,
+                    uploaded_ssts: syncing_data.uploaded,
                     table_watermarks: syncing_data.table_watermarks,
                 }
             });
-            self.add_synced_data(epoch, result);
-            Poll::Ready(Some((epoch, newly_uploaded_sstable_infos)))
+
+            self.add_synced_data(epoch);
+            Poll::Ready(Some((epoch, result)))
         } else {
             Poll::Ready(None)
         }
     }
 
-    /// Poll the success of the oldest spilled task of sealed data. Return `Poll::Ready(None)` if
-    /// there is no spilling task.
-    fn poll_sealed_spill_task(&mut self, cx: &mut Context<'_>) -> Poll<Option<StagingSstableInfo>> {
-        self.sealed_data.spilled_data.poll_success_spill(cx)
-    }
-
     /// Poll the success of the oldest spilled task of unsealed data. Return `Poll::Ready(None)` if
     /// there is no spilling task.
     fn poll_unsealed_spill_task(
         &mut self,
         cx: &mut Context<'_>,
     ) -> Poll<Option<StagingSstableInfo>> {
         // iterator from older epoch to new epoch so that the spill task are finished in epoch order
-        for unsealed_data in self.unsealed_data.values_mut() {
+        for unsealed_data in self.unsync_data.epoch_data.values_mut() {
```

Comment:
I think the term unseal/sealed/sync are used with different meanings in different parts of the codes after this PR. For example, sealed data in `LocalInstanceUnsyncData` is considered unsealed in the context of `HummockUploader`. Can we make it less confusing by having a clear definition of sync and sealed? Some ideas:
- `unsealed`: per instance concept. Data is unsealed before `local_seal_epoch`
- `sealed`: per instance concept. Data is sealed after `local_seal_epoch`
- `spill`: per epoch concept. Data is spilled if `may_spill` triggers.
- `sync`: per epoch conecpt. Data is synced after `sync`.

`unsealed`, `sealed` and `spill` data are all considered `unsync`.

Following this ideas, we can:
1. rename `poll_unsealed_spill_task` to `poll_unsync_spill_task`
2. remove `SealedData` struct and return the corresponding `epoch_data` on `unsync_data.sync`

Reply:
Indeed. I have removed the concept of `sealed` and `unsealed` in global uploader. The concept is only in `LocalInstanceUnsyncData`.